### ISSUE TITLE
Failed to build PDF with xelatex

### ISSUE THREAD

**Issue by tk0miya** [sphinx-doc/sphinx#3746]
### Problem
- Failed to build PDF document using xelatex

#### Procedure to reproduce the problem
```
$ sphinx-quickstart doc
$ cd doc/
$ echo 'latex_engine = "xelatex"' >> conf.py
$ make latexpdf
Running Sphinx v1.6+
making output directory...
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [latex]: all documents
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
processing test.tex...index
resolving references...
writing... done
copying TeX support files...
done
build succeeded.
latexmk -pdfxe -dvi- -ps-  'test.tex'
Latexmk: -pdfxe bad option

Latexmk: Bad options specified
Use
   latexmk -help
to get usage information
make[1]: *** [test.pdf] Error 10
```

#### Expected results
- Succeeded to build PDF.

### Environment info
- OS: Mac
- Python version: 2.7 / 3.6
- Sphinx version: HEAD of 1.6-release branch
- MacTeX 20161009 (installed by homebrew-cask)

**Comment by tk0miya** [comment]
As a quick look, `latexmk` (v4.45) expects `-xelatex` option to build PDF using XeLaTeX.
I succeeded to build PDF by replacing the option in generated Makefile to `-xelatex`.
```
$ cd _build/latex
$ vi Makefile
$ make
latexmk -xelatex -dvi- -ps-  'test.tex'
Latexmk: This is Latexmk, John Collins, 22 April 2016, version: 4.45.
Latexmk: applying rule 'pdflatex'...
Rule 'pdflatex': Rules & subrules not known to be previously run:
   pdflatex
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running 'xelatex  -recorder  "test.tex"'
------------
This is XeTeX, Version 3.14159265-2.6-0.99996 (TeX Live 2016) (preloaded format=xelatex)
 restricted \write18 enabled.

(snip)

Output written on test.pdf (5 pages).
Transcript written on test.log.
Latexmk: Index file 'test.idx' was written
Latexmk: Log file says output to 'test.pdf'
Latexmk: All targets (test.pdf) are up-to-date
```

**Comment by jfbu** [comment]
I am still here ;-) and will give a look now.

**Comment by tk0miya** [comment]
I also got similar error for `latex_engine = "lualatex"`.
```
latexmk -pdflua -dvi- -ps-  'test.tex'
Latexmk: -pdflua bad option

Latexmk: Bad options specified
Use
   latexmk -help
to get usage information
make[1]: *** [test.pdf] Error 10
```
And also fixed by replacing `-pdflua` option by `-lualatex`.

@jfbu  In `sphinx/texinputs/Makefile_t`, the options strings of `LATEX` and `PDFLATEX` are different. Is this intended?
```
{% elif latex_engine == 'lualatex' -%}
LATEX = latexmk -lualatex
PDFLATEX = latexmk -pdflua -dvi- -ps-
{% elif latex_engine == 'xelatex' -%}
LATEX = latexmk -pdfxe -dvi- -ps-
PDFLATEX = $(LATEX)
{% endif %}
```

**Comment by jfbu** [comment]
Yes, for lualatex LATEX produces dvi files, PDFLATEX produces directly pdf files.

For xelatex, intermediate runs by latexmk is much faster because it produces xdv, not pdf files. The pdf is produced on last run. Because there is no viewer for xdv format, there is no difference for LATEX and PDFLATEX for xelatex. We only produce (finally) pdf.

**Comment by tk0miya** [comment]
@jfbu Thanks, but our release is not important so much. Please say "time up" freely and easily. It's not big deal to postpone the release. Nobody hurries us!

**Comment by jfbu** [comment]
I have TL2013 and TL2014 on my computer. I will make sure options for Latexmk are fine there (for final TL2014, it is Latexmk, 5 February 2015. Version 4.43a). I will check with 2013 too.

**Comment by jfbu** [comment]
Documentation is at http://personal.psu.edu/jcc8/software/latexmk/versions.html

Indeed it is big error on my part that I tested only with up-to-date latexmk. (see item 5. in 4.52c)

Will fix.

**Comment by tk0miya** [comment]
The history of latexmk says in `V. 4.52b`:

>Implement direct support for xelatex and lualatex. Previously these programs were used by configuring the $pdflatex so that the desired program is run instead of pdflatex. There are now configuration variables $xelatex and $lualatex to specify the commands used, and extra command line options -pdfxe and -pdflua.
http://personal.psu.edu/jcc8/software/latexmk/versions.html

And there are no descriptions on manuals before v4.52b.

**Comment by tk0miya** [comment]
Synchronicity!? we got same answer at same time :-)

**Comment by jfbu** [comment]
The [latexmk history](http://personal.psu.edu/jcc8/software/latexmk/versions.html) is indeed very useful. ``-lualatex`` option was added in July 2013 (4.37)  Although our doc says we support starting with February 2014 Ubuntu version of TeXLive2013,  I thought it was a bit limiting. So at #3749 I have done it in a way which should work with even older releases.

Now I have tested it works with Latexmk 4.35 of 11 Nov. 2012, both for lualatex and xelatex engine.

We do not support (with 1.6) lualatex on TeXLive 2012, as it is not supported by fontspec: but I did check it does work with replacement of fontspec+polyglossia by babel as for pdflatex engine.
